### PR TITLE
wrangler: Improvements

### PR DESCRIPTION
**Pull request by OilyLime**
**What this PR solves / how to test:**
Adds support for hyperdrive.

**Author has included the following, where applicable:**

- [x] Tests
- [x] Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))

**Reviewer is to perform the following, as applicable:**

- Checked for inclusion of relevant tests
- Checked for inclusion of a relevant changeset
- Checked for creation of associated docs updates
- Manually pulled down the changes and spot-tested

### CODE REVIEW

File: packages/wrangler/src/hyperdrive/create.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+			"You must provide a password - e.g. 'user:password@database.example.com:port/databasename' "
+		);
+	} else {
+		logger.log(`🚧 Creating '${args.name}'`);
+		const database = await createConfig(config, {
+			name: args.name,
+			origin: {
+				host: url.hostname,
+				port: parseInt(url.port),
+				scheme: url.protocol,
```
Comment (elithrar) [inline packages/wrangler/src/hyperdrive/create.ts:72]:
```suggestion
				// Replace the trailing colon: the API rejects it
				scheme: url.protocol.replace(":", "")
```

**Review by jspspike — APPROVED** [review]
Tests and commands look good. Didn't scrutinize the hyperdrive specific logic too closely but the tests make sense and wrangler changes look good

**Review by JacobMGEvans — APPROVED** [review]
Other than the testing not covering the new bindings validation, in config validation.ts it looks good to me.

**Review by RamIdeas — APPROVED** [review]

**Comment by RamIdeas** [comment]
Needs a rebase but LGTM

**Pull request merged**